### ISSUE TITLE
Skip the code 

### ISSUE THREAD

**Issue by reihun** [C-Nedelcu/talk-to-chatgpt#5]
I don't want to hear the code . I want to skip it , just listen description

**Comment by Manamama** [comment]
Thank you: both versions 1.2 and 1.3 work great on the newest Chrome in Ubuntu. 

Some tips about what to include in the Readme for the "GitHub newbies":
* Add the link to [the booklet marker](https://caiorss.github.io/bookmarklet-maker/) that you had in the previous version 1.2. Even better: create the booklet code automatically here, with a clear direct link. 
* Tell them or direct them in the Readme to the Raw JS file. (Not everybody knows how GitHub UI works and what to click if they see it the first time.) 
* As per the request above, add "No output" to the Settings. It is fun and useful to listen in many cases, but gets annoying after a minute or two. 

Otherwise a great tool which works in a couple of languages, subject to Google's STT and TTS capabilities, of course.

**Comment by Manamama** [comment]
Update: I have fed the code to ChatGPT and asked it to update with that missing features ;) 
`Modify this code so that there is a settings option for no voice output. Show me the change: `

The results are more than promising: 
_To add a settings option for no voice output, you can add a new variable called CN_NO_VOICE_OUTPUT and set it to false by default. Then, you can modify the CN_SayOutLoud function to check if CN_NO_VOICE_OUTPUT is true before speaking the text. If it is true, the function can simply return without calling the speechSynthesis.speak function.

Here is an example of how you can make this change:_
[...]
// Determine which word will cause this script to temporarily pause
var CN_SAY_THIS_WORD_TO_PAUSE = "pause";
```
// Set this to true to disable voice output
var CN_NO_VOICE_OUTPUT = true;
```

and then: 

// This function will say the given text out loud using the browser's speech synthesis API
function CN_SayOutLoud(text) {
	if (!text) return;
```
	// Check if no voice output is enabled
	if (CN_NO_VOICE_OUTPUT) return;

```
It also even created a Settings menu item after some gentle prompting ;), but something is messed up and it does not toggle: 

/ Open settings menu
function CN_OnSettingsIconClick() {
[...]
```
// 7. No voice output
var checkbox = document.createElement("input");
checkbox.type = "checkbox";
checkbox.id = "TTGPTNoVoiceOutput";
checkbox.checked = CN_NO_VOICE_OUTPUT;

checkbox.addEventListener("change", function(event) {
  CN_NO_VOICE_OUTPUT = event.target.checked;
});

```
I am tempted to fork it ...

**Comment by C-Nedelcu** [comment]
Haha that is brilliant. I will follow your suggestions. We need an account for ChatGPT so that it can propose changes itself

**Comment by C-Nedelcu** [comment]
In V1.4 I am planning:
- settings persistence
- skip button to stop reading current message
- toggle button for text-to-speech (keeps bot quiet)

**Comment by Manamama** [comment]
Way to go, Nedelcu! Ta for listening to us.
For the time being, I had created two buttons, one with your  talking 1.3 ver. code, the current one, and the second with my (ok, ChatGPT's: me being no coder) 1.3.1 default silence one. 

A further feature request: add wget or like so that  it can be fed by voice with live URLs ;)...  (FYI, I manually tested this by pasting such content manually, and only 20 sentences or like are really fed into AI ChatGPT' episodic memory alas, despite GPT's self declarations that it can remeber all).

**Comment by C-Nedelcu** [comment]
Hello @Manamama  
Can you tell me more about what you mean for your future request regarding wget? I am not sure I understand what you are trying to achieve? Do you want to feed a message that will be sent to the bot ? While that could be useful, it doesn't really fall into the scope of this project which is just to talk to the bot with voice.
Now, there could definitely be a toolbox project to add features to ChatGPT with your ideas and many more.

**Comment by C-Nedelcu** [comment]
Hello,

I have just published v1.4
- settings are now saved in cookie and restored automatically
- revamped menu
- voice recognition toggle
- text-to-speech toggle
- skip current message button
- menu border color changes based on current state (grey=idle, green=speaking, red=listening)
- instead of an alert message, the browser will say 'OK' to indicate that it is ready

I think this means we can close the current issue.
Feel free to create a new thread if you have more features to suggest :)
I will be working on a Chrome extension soon
